admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java

```
@@ -330,6 +351,63 @@
 		return CompletableFuture.completedFuture(machineDetailDtoList);
 	}
 
+	/**
+	 * Method to fetch location hierarchy details
+	 * 
+	 * @param lastUpdated      lastUpdated time-stamp
+	 * @param currentTimeStamp current time stamp
+	 * 
+	 * @return list of {@link LocationHierarchyLevelDto} list of
+	 *         locationHierarchyList dto
+	 */
+	@Async
+	public CompletableFuture<List<LocationHierarchyDto>> getLocationHierarchyList(LocalDateTime lastUpdated,
+			LocalDateTime currentTimeStamp) {
+
+		LocationHierarchyLevelResponseDto locationHierarchyResponseDto = null;
+		List<LocationHierarchyDto> locationHierarchyLevelDtos = new ArrayList<LocationHierarchyDto>();
+
+		final String url = mosipEnvironment.getLocationHierarchyUrl();
+		HttpHeaders headers = new HttpHeaders();
+
+		String lastUpdatedTime = null;
+		UriComponentsBuilder builder = null;
+		if (lastUpdated != null) {
+			DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+			lastUpdatedTime = lastUpdated.format(formatter);
+			builder = UriComponentsBuilder.fromHttpUrl(url).queryParam("lastUpdated", lastUpdatedTime + "Z");
+		} else {
+			builder = UriComponentsBuilder.fromHttpUrl(url);
+		}
+
+		ResponseEntity<String> response = restTemplate.exchange(builder.toUriString(), HttpMethod.GET,
+				new HttpEntity<Object>(headers), String.class);
+		if (response.getStatusCode().equals(HttpStatus.OK)) {
+			String responseBody = response.getBody();
+			List<ServiceError> validationErrorsList = null;
+			validationErrorsList = ExceptionUtils.getServiceErrorList(responseBody);
+			if (!validationErrorsList.isEmpty()) {
+				// throw new SyncServiceException(validationErrorsList);
```

you need to throw this exception